byway/src/main/webapp/track-interests.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+let selected = [];
+
+/** 
+ * Track the selected interests in an array according to their status. 
+ * @param elem tracks the current button element chosen.
+ */
+function checkStatus(elem) {
+  if(elem.className === "btnActive") {
+    selected.push(elem.innerText);
+  } else {
+    selected = selected.filter(function(interest) {
+      return interest !== elem.innerText;
+    });
+  }
+}
+
+/** 
+ * Update the class of the buttons to change their style.
+ * @param elem tracks the current button element chosen.
+ */
+function switchStatus(elem) {
+  let switchClass = (elem.className === "btn"? "btnActive": "btn");
```

Comment:
Note this is part of what I mean about things having multiple classes: there's no reason to have both a `btn` and `btnActive` class, where `btnActive` copies everything about `btn` except the properties which change when its active. It means you end up with a lot of repetition in the CSS, and it's pretty unnecessary, since the "cascading" bit of CSS means you can define a base class like `btn` and then apply additional classes only when necessary:

```css
.btn {
  margin: 10px;
  align-items: center;
  background-color: #DDDDDD;
  border-color: #FFFFFF;
  border-radius: 10px;
  border-width: 1px;
  padding: 10px;
  outline: none;
}

.btn.active {
  background-color: #68d388;
  border-color: #dc00ff;
}
```

Setting both classes, e.g. with `className = 'btn active'` will cause the effects of `btn` to apply then `active` will be applied on top. Note that the selector `.btn.active` in the CSS means that those particular effects of `active` only apply when it is attached to `btn`. You can do this to make `active` have different effects depending on which element its applied to. `.btn .active` (not space) will apply when you have active on a `.btn` or any child element. and `.btn > .active` will apply when you have `active` on on only a child of `.btn` but not the `.btn` element itself.

Reply:
I noticed the repetition as well but wasn't sure how to handle a class with a selector to apply other styles. I initially tried to set it as `btn.active`, but setting it as `btn active` like you mentioned works - thanks. Fixed